bindings/rust/aws-lc-sys-template/build.rs

```diff
@@ -210,59 +196,115 @@ fn find_cmake_command() -> Option<&'static OsStr> {
-        );
+        args.push(OsString::from(format!(
+            "-DBORINGSSL_PREFIX={}",
+            symbol_prefix
+        )));
+        let include_path = aws_lc_path.join("include");
+        args.push(OsString::from(format!(
+            "-DBORINGSSL_PREFIX_HEADERS={}",
+            include_path.display()
+        )));
     }
 
     if cfg!(feature = "asan") {
         env::set_var("CC", "/usr/bin/clang");
         env::set_var("CXX", "/usr/bin/clang++");
         env::set_var("ASM", "/usr/bin/clang");
 
-        cmake_cfg.define("ASAN", "1");
+        args.push(OsString::from("-DASAN=1"));
     }
 
-    cmake_cfg
+    Ok(args)
 }
 
-fn build_aws_lc() -> PathBuf {
-    let mut cmake_cfg = prepare_cmake_build(Some(&prefix_string()));
+fn load_env_var(var_name: &str) -> String {
+    env::var(var_name).unwrap_or_else(|_| panic!("Missing Environment variable: {}", var_name))
+}
+
+fn build_aws_lc() -> Result<PathBuf, &'static str> {
+    let cmake_cmd: &OsStr;
+    if let Some(cmd) = find_cmake_command() {
+        // This can be a let-else as-of Rust 1.65
+        cmake_cmd = cmd;
+    } else {
+        return Err("Missing dependency: cmake");
+    };
 
-    cmake_cfg.build_target("crypto").build()
+    let pwd = env::current_dir().unwrap_or_else(|_| PathBuf::from("."));
+    let aws_lc_path = pwd.join(AWS_LC_PATH);
+
+    let out_dir = PathBuf::from(load_env_var("OUT_DIR"));
+    let build_dir = out_dir.join("build");
+    if !build_dir.exists() {
+        fs::create_dir_all(build_dir.clone()).unwrap();
+    }
+
+    let mut setup_command = Command::new(cmake_cmd);
+
+    setup_command.args(prepare_cmake_args(&aws_lc_path, Some(&prefix_string()))?);
+    setup_command.current_dir(&build_dir);
+    println!("running: {:?}", setup_command);
+    let output = setup_command
+        .output()
+        .map_err(|_| "cmake setup command failed")?;
+
+    if !output.status.success() {
+        eprintln!("{}", String::from_utf8(output.stderr).unwrap());
+        return Err("cmake setup command unsuccessful");
+    }
+
+    let mut build_command = Command::new(cmake_cmd);
+    build_command.args(&[
+        OsString::from("--build"),
+        OsString::from("."),
+        OsString::from("--target"),
+        OsString::from("crypto"),
+    ]);
+    build_command.current_dir(&build_dir);
+
+    println!("running: {:?}", build_command);
+    let output = build_command
+        .output()
+        .map_err(|_| "cmake build command failed")?;
+    if !output.status.success() {
+        eprintln!("-- stdout --");
+        eprintln!("{}", String::from_utf8(output.stdout).unwrap());
+        eprintln!("-- stderr --");
+        eprintln!("{}", String::from_utf8(output.stderr).unwrap());
+        return Err("cmake build command unsuccessful.");
+    }
+
+    Ok(out_dir)
 }
 
 fn main() -> Result<(), String> {
     use crate::OutputLib::Crypto;
     use crate::OutputLibType::Static;
 
     let mut missing_dependency = false;
-    if !test_go_command() {
```

Comment:
just at datapoint: Go is still required when building for FIPS, so we might need to add this back for the FIPS build. But we can remove it for now